### ISSUE TITLE
Add missing API entries to man/toctail

### ISSUE THREAD

**Issue by alycm** [KhronosGroup/OpenCL-Docs#350]
API entry points from OpenCL 2.2 onwards are missing from `man/toctail`:
* `clSetProgramSpecializationConstant`
* `clSetProgramReleaseCallback`
* `clCreateBufferWithProperties`
* `clCreateImageWithProperties`

Note that `clSetContextDestructorCallback` is *not* missing because that was included in the PR that added it.

**Comment by bashbaug** [comment]
I looked at adding `clCreateBufferWithProperties` and `clCreateImageWithProperties` as part of the `clSetContextDestructorCallback` PR, but the way we've documented the "with properties" functions doesn't fit as cleanly into the reference page format as some of the other APIs: we've essentially documented them as a variant of the "non-with-properties" function, so they show up on the reference page for `clCreateBuffer` and `clCreateImage` respectively.

I'm sure there is a way to solve this problem cleanly, but resolving this issue will not be a simple matter of adding these new APIs.

Is there a way to define an "alias" so `clCreateBufferWithProperties` goes to the same page as `clCreateBuffer`, for example?

For the other missing APIs - `clSetPorgramSpecializationConstant` and `clSetProgramReleaseCallback` - these should be more straightforward additions.
